pkg/devspace/build/localregistry/deployment.go

```diff
@@ -58,11 +58,16 @@ func (r *LocalRegistry) ensureDeployment(ctx devspacecontext.Context) (*appsv1.D
 		return nil, err
 	}
 
+	if !ctx.KubeClient().SupportServerSideApply() {
```

Comment:
IIRC, server side apply can also be disabled. Perhaps handling the 415 response and falling back to a Patch operation would be an option that would adapt to more version / configuration combinations?

Reply:
Yes, I proved it, and I would fix it with non-version based approach.
```shell

$ kubectl version
Client Version: v1.30.0
Server Version: v1.18.19

$ kubectl -n kube-system get po -l component=kube-apiserver -o yaml | grep -i serversideapply
      - --feature-gates=ServerSideApply=false

$ kubectl apply --server-side=true -f - <<EOF
> apiVersion: v1
> kind: ConfigMap
> metadata:
>   name: ssa-cm
>   namespace: default
> data:
>   foo: bar
> EOF
error: Server-side apply not available on the server: (415: Unsupported Media Type)
```